src/main/java/it/gov/pagopa/timeline/service/TimelineServiceImpl.java

```diff
@@ -86,12 +96,44 @@ public void sendToQueue(QueueOperationDTO queueOperationDTO) {
   public void saveOperation(QueueOperationDTO queueOperationDTO) {
     long startTime = System.currentTimeMillis();
 
+    if (queueOperationDTO.getChannel().equals(TimelineConstants.CHANNEL_QRCODE)) {
```

Comment:
The channel field is null in case of other operations, so this check would result in a NullPointerException
```suggestion
    if (TimelineConstants.CHANNEL_QRCODE.equals(queueOperationDTO.getChannel())) {
```